src/sync-controller.js

```diff
@@ -29,17 +29,40 @@ export const syncPointStrategies = [
   {
     name: 'ProgramDateTime',
     run: (syncController, playlist, duration, currentTimeline, currentTime) => {
-      if (syncController.datetimeToDisplayTime && playlist.dateTimeObject) {
-        let playlistTime = playlist.dateTimeObject.getTime() / 1000;
-        let playlistStart = playlistTime + syncController.datetimeToDisplayTime;
-        let syncPoint = {
-          time: playlistStart,
-          segmentIndex: 0
-        };
+      if (!syncController.datetimeToDisplayTime) {
+        return null;
+      }
 
-        return syncPoint;
+      let segments = playlist.segments || [];
+      let syncPoint = null;
+      let lastDistance = null;
+
+      currentTime = currentTime || 0;
+
+      for (let i = 0; i < segments.length; i++) {
+        let segment = segments[i];
+
+        if (segment.dateTimeObject) {
+          let segmentTime = segment.dateTimeObject.getTime() / 1000;
+          let segmentStart = segmentTime + syncController.datetimeToDisplayTime;
+          let distance = Math.abs(currentTime - segmentStart);
+
+          // Once the distance begins to increase, we have passed
+          // currentTime and can stop looking for better candidates
+          if (lastDistance !== null && lastDistance < distance) {
+            break;
+          }
+
+          if (!syncPoint || lastDistance === null || lastDistance >= distance) {
```

Comment:
Do we need this check here?

Reply:
Copied this logic from the Segment strategy below, but looking it over again, it doesn't look like it.